spec/github/message_builder_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'spec_helper'
+
+describe MiqToolsServices::Github::MessageBuilder do
+  let(:max) { described_class::GITHUB_COMMENT_BODY_MAX_SIZE }
+  let(:header) { "header stuff\n\n" }
+  let(:continuation_header) { "continued...\n\n" }
+  subject { described_class.new(header, continuation_header) }
+
+  describe "#write / #messages" do
+    it "with simple line" do
+      subject.write("a line")
+      expect(subject.messages).to eq ["#{header}a line\n"]
+    end
+
+    it "with a line that's too long" do
+      expect { subject.write("a" * max) }.to raise_error(ArgumentError)
+    end
+
+    it "with a line that will create a new message" do
+      line = "a" * (max - header.length)
+      subject.write("a line")
+      subject.write(line)
+      expect(subject.messages).to eq [
+        "#{header}a line\n",
+        "#{continuation_header}#{line}\n"
+      ]
+    end
+  end
+
+  describe "#write_lines / #messages" do
+    it "with simple lines" do
+      subject.write_lines(["a line", "another line"])
+      expect(subject.messages).to eq ["#{header}a line\nanother line\n"]
+    end
+
+    it "with a line that's too long" do
+      expect { subject.write_lines(["a line", "a" * max]) }.to raise_error(ArgumentError)
+    end
+
+    it "with a line that will create a new message" do
```

Comment:
Not sure what how to phrase this but "wraps to a new message" seems better to me... not really... can't really come up with anything better.